Source/CarthageKit/FrameworkExtensions.swift

```diff
@@ -154,3 +155,72 @@ extension NSScanner {
 		return nsString.substringWithRange(lineRange)
 	}
 }
+
+extension NSURLSession {
+	/// Returns a signal that will download a file using the given request. The
+	/// file will be deleted after the URL has been sent upon the signal.
+	internal func rac_downloadWithRequest(request: NSURLRequest) -> ColdSignal<(NSURL, NSURLResponse)> {
+		return ColdSignal { (sink, disposable) in
+			let serialDisposable = SerialDisposable()
+			let handle = disposable.addDisposable(serialDisposable)
+
+			let task = self.downloadTaskWithRequest(request) { (URL, response, error) in
+				// Avoid invoking cancel(), or the download may be deleted.
+				handle.remove()
+
+				if URL == nil || response == nil {
+					sink.put(.Error(error))
+				} else {
+					let value = (URL!, response!)
+					sink.put(.Next(Box(value)))
+					sink.put(.Completed)
+				}
+			}
+
+			serialDisposable.innerDisposable = ActionDisposable {
+				task.cancel()
+			}
+
+			task.resume()
+		}
+	}
+}
+
+extension NSURL: JSONDecodable {
+	public class func decode(json: JSONValue) -> Self? {
+		if let URLString = String.decode(json) {
+			return self(string: URLString)
+		} else {
+			return nil
+		}
```

Comment:
I think
```objc
switch j {
        case .JSONString(let string):
            return NSURL(string: string)
        default:
            return .None
}
```
is slightly more elegant but that's only because I wrote this exact code yesterday.